oak_sign/src/lib.rs

```diff
@@ -31,44 +31,59 @@ pub const PUBLIC_KEY_TAG: &str = "PUBLIC KEY";
 pub const SIGNATURE_TAG: &str = "SIGNATURE";
 pub const HASH_TAG: &str = "HASH";
 
-/// Convenience struct that encapsulates `Ed25519KeyPair`.
-/// Named `KeyBundle` since it uses the `ring::signature::KeyPair` trait.
-pub struct KeyBundle {
-    private_key: Vec<u8>,
-    key_pair: Ed25519KeyPair,
+/// Convenience struct that encapsulates `ring::signature::Ed25519KeyPair`.
+pub struct KeyPair {
+    /// PKCS#8 v2 encoded private key and public key.
+    /// https://tools.ietf.org/html/rfc5958
+    ///
+    /// The encoded version is kept because a `parsed` key pair doesn't contain a method for
+    /// extracting private key bytes.
+    /// https://briansmith.org/rustdoc/ring/signature/struct.Ed25519KeyPair.html
+    encoded_key_pair: Vec<u8>,
+    /// Parsed PKCS#8 v2 key pair representation.
+    parsed_key_pair: Ed25519KeyPair,
 }
 
-impl KeyBundle {
+impl Clone for KeyPair {
+    fn clone(&self) -> Self {
+        KeyPair::parse(&self.encoded_key_pair).expect("Couldn't clone key pair")
+    }
+}
+
+impl KeyPair {
     /// Generates a Ed25519 key pair.
-    pub fn generate() -> anyhow::Result<KeyBundle> {
+    pub fn generate() -> anyhow::Result<KeyPair> {
         let rng = rand::SystemRandom::new();
         let private_key = Ed25519KeyPair::generate_pkcs8(&rng)
             .ok()
             .context("Couldn't generate key pair")?;
-        KeyBundle::parse(private_key.as_ref())
+        KeyPair::parse(private_key.as_ref())
     }
 
     /// Parses a Ed25519 key pair from bytes.
-    pub fn parse(private_key: &[u8]) -> anyhow::Result<KeyBundle> {
-        let key_pair = Ed25519KeyPair::from_pkcs8(private_key)
+    pub fn parse(private_key: &[u8]) -> anyhow::Result<KeyPair> {
```

Comment:
Could you clarify that the bytes are pkcs8, ideally both in the param name, and in the comment? Otherwise it seems impossible for anyone to use this method (or maybe it should not be public?)